cabal-install/src/Distribution/Client/Init/Format.hs

```diff
@@ -109,9 +109,11 @@ annNoComments :: FieldAnnotation
 annNoComments = FieldAnnotation False NoComment
 
 postProcessFieldLines :: FieldAnnotation -> [String] -> [String]
-postProcessFieldLines ann
-    | annCommentedOut ann = fmap ("-- " ++)
-    | otherwise = id
+postProcessFieldLines ann comments
+  | annCommentedOut ann = case () of
```

Comment:
BTW, you can express this without the vacuous case over  `()`. e.g., with normal if or MultiWayIf (though hlint would complain there are only 2 cases for MultiWayIf).